extensions/vscode-api-tests/src/workspace.test.ts

```diff
@@ -160,6 +160,41 @@ suite('workspace-namespace', () => {
 		});
 	});
 
+	test('eol, read', () => {
+		const a = createRandomFile('foo\nbar\nbar').then(file => {
+			return workspace.openTextDocument(file).then(doc => {
+				assert.equal(doc.eol, EndOfLine.LF);
+			});
+		});
+		const b = createRandomFile('foo\nbar\nbar\r\nbaz').then(file => {
+			return workspace.openTextDocument(file).then(doc => {
+				assert.equal(doc.eol, EndOfLine.LF);
+			});
+		});
+		const c = createRandomFile('foo\r\nbar\r\nbar').then(file => {
+			return workspace.openTextDocument(file).then(doc => {
+				assert.equal(doc.eol, EndOfLine.CRLF);
+			});
+		});
+		return Promise.all([a, b, c]);
+	});
+
+	test('eol, change via editor', () => {
+		return createRandomFile('foo\nbar\nbar').then(file => {
+			return workspace.openTextDocument(file).then(doc => {
+				assert.equal(doc.eol, EndOfLine.LF);
+				return window.showTextDocument(doc).then(editor => {
+					return editor.edit(builder => builder.setEndOfLine(EndOfLine.CRLF));
```

Comment:
If so, what will happen in a "Save All" scenario? Will each editor show for a split second before applying `setEndOfLine`? Doesn't seem ideal. Any way to do this with a `TextEdit` for `onWillSaveTextDocument`?